### ISSUE TITLE
Assertion failed during 'Mark local vars'

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#43508]
Assertion failed '(varDsc->lvType != TYP_UNDEF) && (varDsc->lvType != TYP_VOID) && (varDsc->lvType != TYP_UNKNOWN)' in 'FullProof:Test():int' during 'Mark local vars'

JIT/Regression/VS-ia64-JIT/V2.0-Beta2/b309576/b309576

category:correctness
theme:needs-triage
skill-level:beginner
cost:small

**Comment by BruceForstall** [comment]
@MichalStrehovsky How do you repro? I removed `<Optimize>True</Optimize>` from b309576.ilproj, rebuilt the test, then ran:
```
> c:\gh\runtime\artifacts\tests\coreclr\Windows_NT.x64.Debug\tests\Core_Root\CoreRun.exe C:\gh\runtime\artifacts\tests\coreclr\Windows_NT.x64.Debug\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b309576\b309576\b309576.dll
System.OverflowException: Arithmetic operation resulted in an overflow.
   at FullProof.Test()
   at FullProof.Main()
PASS
```

**Comment by MichalStrehovsky** [comment]
I first hit it in runtimelab because I had a change to infrastructure that was ignoring this flag.

The repro in the runtime repo (and the associated jitdump) is based on simply running `ilasm /debug bug2.il` and running corerun on that.

**Comment by MichalStrehovsky** [comment]
Oh, now I see in jitdump:

OPTIONS: compCodeOpt = BLENDED_CODE
OPTIONS: compDbgCode = true
OPTIONS: compDbgInfo = true
OPTIONS: compDbgEnC  = false

Is this about optimized debuggable code?

**Comment by jkotas** [comment]
This repros it for me in current dotnet/runtime:master
```
build -rc checked
cd artifacts\bin\coreclr\Windows_NT.x64.Checked
copy D:\runtime\artifacts\bin\runtime\net6.0-Windows_NT-Debug-x64
copy D:\runtime\src\tests\JIT\Regression\VS-ia64-JIT\V2.0-Beta2\b309576
ilasm /DEBUG bug2.il
corerun bug2.exe
```

**Comment by JulieLeeMSFT** [comment]
CC @jakobbotsch PTAL.

**Comment by jakobbotsch** [comment]
I cannot reproduce this. Looks like it was fixed by #48152. After reverting that, I am able to reproduce it.
